ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/config_manager.py

```diff
@@ -109,12 +114,19 @@ def _get_loopback_mappings(self) -> dict:
             # If the host is a VTEP, add the VTEP IP to the mapping
             vtep_interface = get(host_struct_cfg, "vxlan_interface.Vxlan1.vxlan.source_interface")
 
-            # NOTE: For now we exclude WAN VTEPs from the vtep_mapping
             if vtep_interface is not None and "Dps" not in vtep_interface:
                 if (loopback_interface := get_item(loopback_interfaces, "name", vtep_interface)) is None:
                     LOGGER.warning("Host '%s' interface '%s' is missing.", host, vtep_interface)
                 elif (loopback_ip := loopback_interface.get("ip_address")) is None:
                     LOGGER.warning("Host '%s' variable 'ip_address' of interface '%s' is missing.", host, vtep_interface)
                 else:
                     results["vtep_mapping"].append((host, str(ip_interface(loopback_ip).ip)))
+            if vtep_interface is not None and "Dps" in vtep_interface:
```

Comment:
the two ifs logic can be rewritten in a clean fashion - maybe even with a helper funciton

```
if vtep_interface is None:
  continue
if "Dps" in vtep_interfacs:
  _get_ip_address(dps_interfaces, vtep_interface)
else:
  _get_ip_address(loopback_interfaces, vtep_interface)
  ```

Reply:
updated